sdk/identity/azure-identity/TOKEN_CACHING.md

```diff
@@ -0,0 +1,94 @@
+
+The in-memory token cache provided by the Azure Identity library can be used by multiple threads concurrently.
+
+**Note:** When Azure Identity library credentials are used with Azure service libraries (for example, Azure Blob Storage), the in-memory token caching is active in the `HttpPipeline` layer as well. All `TokenCredential` implementations are supported there, including custom implementations external to the Azure Identity library.
+
+### Persistent token caching
+
+*Persistent disk token caching* is an opt-in feature in the Azure Identity library. The feature allows apps to cache access tokens in an encrypted, persistent storage mechanism. As indicated in the following table, the storage mechanism differs across operating systems.
+
+| Operating system | Storage mechanism |
+|------------------|-------------------|
+| Linux            | Keyring           |
+| macOS            | Keychain          |
+| Windows          | DPAPI             |
+
+With persistent disk token caching enabled, the library first determines if a valid access token for the requested resource is already stored in the persistent cache. If a valid token is found, it's returned to the app without the need to make another request to Azure AD. Additionally, the tokens are preserved across app runs, which:
+
+- Makes the app more resilient to failures.
+- Ensures the app can continue to function during an Azure AD outage or disruption.
+- Avoids having to prompt users to authenticate each time the process is restarted.
+
+#### Code sample
+
+The sample showcases how to activate persistence token caching in the credentials offered by the Azure Identity library. You need to specify `TokenCachePersistenceOptions` when creating the credential to activate persistent token caching.
+
+```python 
+ClientSecretCredential(
+    "tenant", "client-id", "secret", cache_persistence_options=TokenCachePersistenceOptions()
+)
+```
+
+#### Persist user authentication record
+When authenticating a user via `InteractiveBrowserCredential`, `DeviceCodeCredential`, or `UsernamePasswordCredential`, an `AuthenticationRecord` can be persisted as well. The authentication record is:
+- Returned from the `authenticate` API and contains data identifying an authenticated account.
+- Needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions.
+There's no sensitive data in the `AuthenticationRecord`, so it can be persisted in a non-protected state.
+Here's an example of an app storing the `AuthenticationRecord` to the local file system after authenticating the user:
```

Comment:
```suggestion
#### Persist user authentication record

When authenticating a user via `InteractiveBrowserCredential`, `DeviceCodeCredential`, or `UsernamePasswordCredential`, an `AuthenticationRecord` can be persisted as well. The authentication record is:

- Returned from the `authenticate` API and contains data identifying an authenticated account.
- Needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions.

There's no sensitive data in the `AuthenticationRecord`, so it can be persisted in a non-protected state. Here's an example of an app storing the `AuthenticationRecord` to the local file system after authenticating the user:
```